app/page/networkList/index.tsx

```diff
@@ -1,19 +1,60 @@
+import { Wallet } from "ethers"
+import { useContext } from "react"
 import Circle from "react:~assets/circle.svg"
 import CircleDot from "react:~assets/circleDot.svg"
 import { useHashLocation } from "wouter/use-hash-location"
 
 import { StepBackHeader } from "~app/component/stepBackHeader"
-import { useAction, useNetwork, useNetworks } from "~app/hook/storage"
+import { ProviderContext } from "~app/context/provider"
+import { ToastContext } from "~app/context/toastContext"
+import {
+  useAction,
+  useNetwork,
+  useNetworks,
+  useTotalAccountCount
+} from "~app/hook/storage"
 import { Path } from "~app/path"
+import { AccountType } from "~packages/account"
+import { SimpleAccount } from "~packages/account/SimpleAccount"
+import number from "~packages/util/number"
 
 export function NetworkList() {
   const [, navigate] = useHashLocation()
-  const { switchNetwork } = useAction()
+  const { provider } = useContext(ProviderContext)
+  const { setToast } = useContext(ToastContext)
+
+  const { switchNetwork, createSimpleAccount } = useAction()
 
   const network = useNetwork()
   const networks = useNetworks()
+  const totalAccountCount = useTotalAccountCount()
+
   const selectNetwork = async (networkId: string) => {
+    // Switch networks first to update the provider to the target network
     await switchNetwork(networkId)
+
+    const targetNetwork = networks.find((network) => network.id === networkId)
+
+    // Initialize an account if it doesn’t exist on the target network.
+    if (
+      targetNetwork &&
+      !targetNetwork?.accountActive &&
+      targetNetwork.accountFactory[AccountType.SimpleAccount] &&
+      targetNetwork.accountFactory[AccountType.SimpleAccount] !== "0x"
```

Comment:
Are you referring to [this handling](https://github.com/pilagod/waallet-extension/blob/6d834679bb392e91dc433302ea5a908df4004f7f/storage/local/manager.ts#L21-L31  )?  

Since the factory address [might be "0x"](https://github.com/pilagod/waallet-extension/blob/da1131282ccf6dcb5ecf69c86857fc888def1624/config/development/index.ts#L34), We should determine the best way to handle this...

Reply:
It covers both scenarios: `SimpleAccount.init` without a factory and `SimpleAccount.initWithFactory` with a factory.
In our case, a factory is required for successful deployment of the `SimpleAccount` contract."